cpp/src/strings/filling/fill.cu

```diff
@@ -46,10 +46,12 @@ std::unique_ptr<column> fill( strings_column_view const& strings,
     if( begin==end ) // return a copy
         return std::make_unique<column>( strings.parent() );
 
-    auto execpol = rmm::exec_policy(stream);
-    string_view d_value(nullptr,0);
-    if( value.is_valid() )
-        d_value = string_view(value.data(),value.size());
+    // string_scalar.data() is null for valid, empty strings
+    string_view const d_value = [&] {
+        if( !value.is_valid() )
+            return string_view(nullptr,0);
+        return value.size()==0 ? string_view("",0) : string_view(value.data(),value.size());
+    } ();
```

Comment:
Why is this a lambda?

Reply:
This is called an immediately initializing lambda expression (IILE). You can use it to initialize a variable without the initialize-update pattern. Mainly it allows the variable `d_value` to be `const`.